clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -1665,6 +1666,22 @@ public static ConfigDef mergeConfigs(List<ConfigDef> configDefs) {
         configDefs.forEach(configDef -> configDef.configKeys().values().forEach(all::define));
         return all;
     }
+
+    /**
+     * Generates a unique connection ID for the given socket.
+     *
+     * @param socket The socket for which the connection ID is to be generated.
+     * @param nextConnectionIndex The index to be used in the connection ID to ensure uniqueness.
+     * @return A string representing the unique connection ID.
+     */
+    public static String generateConnectionId(Socket socket, int nextConnectionIndex) {
```

Comment:
nit: in this context, it can just be connectionIndex

Reply:
Yes, the method don't care about how to generate index.